src/includes/class_reservationPermissions.php

```diff
@@ -1,76 +1,62 @@
 <?php
 class reservationPermissions {
 
-  public function getRecords($id = null){
-      try {
-          // call engine
-          $engine    = EngineAPI::singleton();
-          $localvars = localvars::getInstance();
-          $db        = db::get($localvars->get('dbConnectionName'));
-          $sql       = "SELECT * FROM `reservePermissions`";
-          $validate  = new validate;
+  private $localvars;
+  private $engine;
+  private $this->db;
+  private $validate;
+
+  function __construct() {
+    $this->engine    = EngineAPI::singleton();
+    $this->localvars = localvars::getInstance();
+    $this->db        = db::get($this->localvars->get('dbConnectionName'));
+    $this->validate  = new validate;
+  }
 
-          if (isset($id)) {
-           $id        = dbSanitize($id);
-          }
+  public function getRecords($id = null){
+    $sql = "SELECT * FROM `reservePermissions`";
+    try {
+      if(!isnull($id) && !$this->validate->integer($id)){
```

Comment:
Switched the conditional ordering. We should check if it exists/not null BEFORE we check if it is valid. PHP will fail the conditional at the null check and not waste the CPU on the second check. 

if you do the second check first you run the risk of trying to check null (in this case not an issue, in others it could change the results drastically)